Yoast/wordpress-seo
# Create SEMrush Country Selection component

## Context


* Creates the SEMrush Country Selection component as shown in the images below. 

<img width="400" alt="dropdown" src="https://user-images.githubusercontent.com/43582255/88031318-b72edf00-cb3c-11ea-9a2f-468ba2a9ee00.png"> 
<img width="350" alt="dropdown-open" src="https://user-images.githubusercontent.com/43582255/88031313-b433ee80-cb3c-11ea-8e89-b15f825d69fe.png">


## Summary


This PR can be summarized in the following changelog entry:

* Creates the SEMrush Country Selection component. 

## Relevant technical choices:

* The `CountrySelector` component is prefixed with `SemRush`.
* The dropdown uses Select2 (used the `MultiSelect` component in `javascript/packages/components/src/select/Select.js` as reference). 
* The database country is saved in the state of the `SemRushCountrySelector`.

For the CR: please also carefully look at the list of countries/databases. See the [SEMrush documentation](https://www.semrush.com/api-analytics/#databases) for reference.

**Note**: the "Change Country" button has no functionality yet. This will be handled in a different issue together with storing and retrieving the country from the database. 

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch and build
* Open the SEMrush modal
* Make sure the CountrySelector has the correct styling
* Open the CountrySelector and select a different country
* Check whether the state changed
* Check whether the search function works properly
* Check anything else you come up with :) 

## UI changes

## Documentation

## Quality assurance


Fixes [P3-40]
